dbms/src/Client/PerformanceTest.cpp

```diff
@@ -0,0 +1,1556 @@
+        number_of_bytes_speed_info_batches = 0;
+    }
+};
+
+double Stats::avg_rows_speed_precision  = 0.001;
+double Stats::avg_bytes_speed_precision = 0.001;
+
+class PerformanceTest
+{
+public:
+    using Strings = std::vector<String>;
+
+    PerformanceTest(
+        const String & host_,
+        const UInt16 port_,
+        const String & default_database_,
+        const String & user_,
+        const String & password_,
+        const bool & lite_output_,
+        const String & profiles_file_,
+        Strings && input_files_,
+        Strings && tests_tags_,
+        Strings && skip_tags_,
+        Strings && tests_names_,
+        Strings && skip_names_,
+        Strings && tests_names_regexp_,
+        Strings && skip_names_regexp_
+    )
+        : connection(host_, port_, default_database_, user_, password_),
+          gotSIGINT(false),
+          lite_output(lite_output_),
+          profiles_file(profiles_file_),
+          input_files(input_files_),
+          tests_tags(std::move(tests_tags_)),
+          skip_tags(std::move(skip_tags_)),
+          tests_names(std::move(tests_names_)),
+          skip_names(std::move(skip_names_)),
+          tests_names_regexp(std::move(tests_names_regexp_)),
+          skip_names_regexp(std::move(skip_names_regexp_))
+    {
+        if (input_files.size() < 1)
+        {
+            throw DB::Exception("No tests were specified", 0);
+        }
+
+        std::cerr << std::fixed << std::setprecision(3);
+        std::cout << std::fixed << std::setprecision(3);
+        processTestsConfigurations(input_files);
+    }
+
+private:
+    unsigned concurrency;
+    String test_name;
+
+    using Query = String;
+    using Queries = std::vector<Query>;
+    using QueriesWithIndexes = std::vector<std::pair<Query, size_t>>;
+    Queries queries;
+
+    Connection connection;
+
+    using Keys = std::vector<String>;
+
+    Settings settings;
+
+    InterruptListener interrupt_listener;
+
+    double average_speed_precision = 0.001;
+
+    using XMLConfiguration = Poco::Util::XMLConfiguration;
+    using AbstractConfig = Poco::AutoPtr<Poco::Util::AbstractConfiguration>;
+    using Config = Poco::AutoPtr<XMLConfiguration>;
+
+    using Paths = std::vector<String>;
+    using StringToVector = std::map<String, std::vector<String>>;
+    StringToVector substitutions;
+
+    using StringKeyValue = std::map<String, String>;
+    std::vector<StringKeyValue> substitutions_maps;
+
+    bool gotSIGINT;
+    std::vector<StopCriterions> stop_criterions;
+    String main_metric;
+    bool lite_output;
+    String profiles_file;
+
+    Strings input_files;
+    std::vector<Config> tests_configurations;
+
+    Strings tests_tags;
+    Strings skip_tags;
+    Strings tests_names;
+    Strings skip_names;
+    Strings tests_names_regexp;
+    Strings skip_names_regexp;
+
+    #define incFulfilledCriterions(index, CRITERION)                                                                        \
```

Comment:
Macros should not looks like a regular function.
It should has CAPS name, be declared directly before it usage and be `#undef`ined after last usage.